### PR TITLE
Add facesearch client

### PR DESCRIPTION
**Pull request by alexmorten**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Cleanup (changes in structure but not functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Go version:
* Kafka version:
* Kubernetes version:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: elastic/search/facetest/main.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	"google.golang.org/grpc"
+)
+
+func main() {
+
+	if len(os.Args) != 2 {
+		panic("requires exactly one param - the url to an image")
+	}
+
+	con, err := grpc.Dial("localhost:6666", grpc.WithInsecure())
```
Comment (1Jo1) [inline elastic/search/facetest/main.go]: env variable for the port would be nice

File: elastic/search/faces/face_search_client.go [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+	}
+	faces := response.Faces
+	log.Println("found ", len(faces), " faces in given image")
+	foundFaces := []FoundFace{}
+	for _, face := range faces {
+		r, err := c.searchForSimilarFaces(face, maxHitsPerFace)
+		if err != nil {
+			return nil, err
+		}
+		fmt.Println("max score:", r.Hits.MaxScore)
```
Comment (1Jo1) [inline elastic/search/faces/face_search_client.go]: log

File: elastic/search/facetest/main.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+	if err != nil {
+		panic(err)
+	}
+	faceReconClient := proto.NewFaceRecognizerClient(con)
+	cfg := elasticsearch.Config{Addresses: []string{"http://localhost:9200"}}
+	esClient, err := elasticsearch.NewClient(cfg)
```
Comment (Urhengulas) [inline elastic/search/facetest/main.go]: Use [InitializeElasticSearch](https://github.com/codeuniversity/smag-mvp/blob/master/elastic/utils.go#L8) here

File: elastic/search/faces/face_search_client.go [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+)
+
+// FaceSearchClient searches elasticsearch for similar faces to the faces given in an image
+type FaceSearchClient struct {
+	FaceRecognitionClient proto.FaceRecognizerClient
+	ESClient              *elasticsearch.Client
+}
+
+// FindSimilarFacesInImage finds faces in the given image and searches for faces similar to the ones in the given image
+// the url should be downloadable be the face recognizer, i.e. a signed url for s3 or for the imgproxy
```
Comment (Urhengulas) [inline elastic/search/faces/face_search_client.go]:
```suggestion
// the url should be downloadable by the face recognizer, i.e. a signed url for s3 or for the imgproxy
```

**Review by 1Jo1 — APPROVED** [review]

**Review by Urhengulas — DISMISSED** [review]

**Pull request merged**